docs/madr/decisions/073-spiffe-compliance.md

```diff
@@ -153,7 +157,919 @@ Future releases
+> https://spiffe.io/docs/latest/deploying/spire_agent/#sds-configuration
+> https://spiffe.io/docs/latest/microservices/envoy/#tls-certificates
+
+**Spire**
+
+In the case of Spire, all certificates are delivered by Spire’s SDS server. The CA configured in `MeshTrust` is not delivered or used for workload identity issuance. However, if the user wants to enable federation with another trust domain, they are required to create a `MeshTrust` resource that defines the `trustDomain` of the federated cluster — but without specifying a CA. This is because the CA will be provided dynamically by Spire as part of the federation mechanism.
+
+#### MeshMultiZoneService
+
+A `MeshMultiZoneService` is composed of multiple MeshService instances, potentially spanning different zones that may each use different identity providers.
+To ensure that cross-zone traffic functions correctly in this setup, we need to:
+* Set the appropriate Subject Alternative Names (SANs) in the Clusters, and
+* Configure the correct list of trusted Certificate Authorities (CAs).
+
+This ensures that workloads can authenticate each other across zones using mTLS, even when different trust domains are involved.
+
+### Migration from mTLS in Mesh to MeshIdentity
+
+1. Mesh has mTLS enabled
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: Mesh
+metadata:
+  name: default
+spec:
+  mtls:
+    enabledBackend: ca-1
+    backends:
+      - name: ca-1
+        type: builtin
+```
+2. User create a new MeshIdentity with example service to verify
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshIdentity
+metadata:
+  name: identity
+  namespace: kuma-system # only in system namespace
+  labels:
+    kuma.io/mesh: default
+spec:
+  selector:
+    dataplane:
+      matchLabels:
+        app: demo-app
+  spiffeID: # optional
+    trustDomain: "prod.zone-1.mesh.local"
+    path: "/ns/{{ .Namespace }}/sa/{{ .ServiceAccount }}"
+  bundled: # to extend in KM
+    meshTrustCreation: Disabled | Enabled # allows to disable extraction of CA into MeshTrust
+    insecureAllowSelfSigned: true
+    certificateParameters: # lets make rotation percentage static for the moment
+      expiry: 24h
+    autogenerate:
+      enabled: true
+    ca:
+      certificate:
+        type: File | Secret | EnvVar
+        # OneOf
+        secret:
+        file:
+        envVar:
+      privateKey:
+        type: File | Secret | EnvVar
+        # OneOf
+        secret:
+        file:
+        envVar:
+```
+
+In the meantime the `MeshTrust` is created and propagated to each Dataplane.
+3. Create MeshTrafficPermissions for accept all traffic
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshTrafficPermission
+metadata:
+  name: allow-all
+  namespace: kuma-system
+  labels:
+    kuma.io/mesh: default
+spec:
+  from:
+  - targetRef:
+      kind: Mesh
+    default:
+      action: Allow
+```
+or more specific but it's not a part of this MADR
```

Comment:
Is it going to be defined in https://github.com/kumahq/kuma/pull/13848 ?

Reply:
Yes